django/db/models/sql/query.py

```diff
@@ -2065,3 +1924,219 @@ def update_join_types(self, query):
+            if lookups_list[-1] not in ('exact', 'iexact'):
+                raise ValueError("Cannot use None as a query value")
+            lookups_list[-1] = 'isnull'
+            value = True
+        elif hasattr(value, 'resolve_expression'):
+            pre_joins = self.query.alias_refcount.copy()
+            value = value.resolve_expression(self.query, reuse=can_reuse, allow_joins=allow_joins)
+            used_joins = [k for k, v in self.query.alias_refcount.items() if v > pre_joins.get(k, 0)]
+        # Subqueries need to use a different set of aliases than the
+        # outer query. Call bump_prefix to change aliases of the inner
+        # query (the value).
+        if hasattr(value, 'query') and hasattr(value.query, 'bump_prefix'):
+            value = value._clone()
+            value.query.bump_prefix(self.query)
+        if hasattr(value, 'bump_prefix'):
+            value = value.clone()
+            value.bump_prefix(self.query)
+        # For Oracle '' is equivalent to null. The check needs to be done
+        # at this stage because join promotion can't be done at compiler
+        # stage. Using DEFAULT_DB_ALIAS isn't nice, but it is the best we
+        # can do here. Similar thing is done in is_nullable(), too.
+        if (connections[DEFAULT_DB_ALIAS].features.interprets_empty_strings_as_nulls and
+                lookups_list[-1] == 'exact' and value == ''):
+            value = True
+            lookups_list[-1] = 'isnull'
+        return value, lookups_list, used_joins
+
+    def build_lookup(self, lookups_list, lhs, rhs):
+        """
+        Tries to extract transforms and lookup from given lhs.
+
+        The lhs value is something that works like SQLExpression.
+        The rhs value is what the lookup is going to compare against.
+        The lookups is a list of names to extract using get_lookup()
+        and get_transform().
+        """
+        lookups_list = lookups_list[:]
+        while lookups_list:
+            name = lookups_list[0]
+            # If there is just one part left, try first get_lookup() so
+            # that if the lhs supports both transform and lookup for the
+            # name, then lookup will be picked.
+            if len(lookups_list) == 1:
+                final_lookup = lhs.get_lookup(name)
+                if not final_lookup:
+                    # We didn't find a lookup. We are going to interpret
+                    # the name as transform, and do an Exact lookup against
+                    # it.
+                    lhs = self.query.try_transform(lhs, name, lookups_list)
+                    final_lookup = lhs.get_lookup('exact')
+                return final_lookup(lhs, rhs)
+            lhs = self.query.try_transform(lhs, name, lookups_list)
+            lookups_list = lookups_list[1:]
+
+
+class QueryObjectLookupHelper(QueryKeywordLookupHelper):
+        def solve_lookup_type(self, lookup):
+            field = lookup.get_source_f_object()
```

Comment:
Actually, it was the most doubtful solution. However, lookup support only F object, transforms or funcs as lhs param. Currently they all support only a single output field (am I right?), so we are sure, lhs contains only a single F object.